slackbridge/messages.py

```diff
@@ -55,7 +57,52 @@ def resolve(self):
         if not channel_id or not isinstance(channel_id, str):
             return
 
-        if channel_id in self.bridge_bot.channels:
+        if channel_id[0] == 'D':  # DM channels start with a D
+            if not user_bot.im_id:
+                user_bot.im_id = channel_id
+
+            if 'text' in self.raw_message:
+                match = re.search('(.*\w):', self.raw_message['text'])
+                if match:
+                    rcpt = match.group(1)
+                    if rcpt in self.bridge_bot.irc_users and self.deferred:
+                        irc_user = self.bridge_bot.irc_users[rcpt]
+                        if irc_user.authenticated:
+                            self.raw_message['text'] = \
+                                self.raw_message['text']\
+                                .replace(match.group(0), '')\
+                                .strip()
+
+                            self._post_pm_to_irc(rcpt, user_bot)
+                            return
+                        else:
+                            self.bridge_bot.post_to_slack(
+                                self.bridge_bot.nickname,
+                                channel_id,
+                                'Error: ' + rcpt + ' is '
```

Comment:
I'd declare the string before this. The multiline string inside the method hurts readability.